docs/proposals/service-catalog.md

```diff
@@ -0,0 +1,408 @@
+and sales.  The IT department wants the marketing department to be able to make
+as many unique usernames and passwords as it wants, the shipping department to
+be able to have up to 25 users, and the sales department to have 5 users.  The
+IT department should be able to write a policy for the database service that
+expresses these requirements.  So, when a user from the shipping department
+tries to consume the service for the 26th time, they receive an error and have
+to contact IT to have the policy changed so that they can get the additional
+user they need.
+
+### Metrics for operators
+
+Operators naturally want to be able to see usage information for their catalog
+entries.  Understanding how many users are consuming the entry helps service
+operators and recipe creators understand the business impact of their catalog
+entries.  Usage data is also necessary to construct market places where users
+are charged for their use of a particular catalog entry.
+
+### Metrics for service consumers
+
+Consumers of entries from the catalog want to be able to see exactly which
+entries their projects consume.  For example, if there is a cost associated with
+using a catalog entry, a user will want to be able to ensure that their project
+consumes only the catalog entries that are necessary for their project to
+function.
+
+### Provisioning resources
+
+A number of use-cases that have already been discussed involve provisioning new
+resources in Kubernetes.  It should also be possible to provision new resources
+outside the cluster when an entry in the catalog is consumed.  For example, in
+the case of an external database, the service operator may want to create a new
+tablespace in the database for each user when the user adds the entry for the
+database to their project.  It should be possible, therefore, for the author of
+a service or recipe to write a custom workflow that is invoked when a user
+consumes their catalog entry.
+
+## Prior Art
+
+### Helm
+
+[Helm](https://github.com/kubernetes/helm) uses “charts” (e.g. resource
+templates) that can instantiated as “releases”. Charts are stored in a HTTP
+repository as compressed tarballs.  The tarball contains, among other files:
+
+- `chart.yaml` - metadata about the chart
+- `values.yaml` -  default variable values to use when processing the resource
+   template files
+- `templates/` - Kubernetes resource templates processed at `helm install` time
+
+Workflow:
+
+- `helm init` -  starts Tiller in the cluster and initializes CLI environment.
+   Tiller runs in the cluster and manages the running Releases.
+- `helm install <chartname>` - downloads the chart, processes the resource
+   templates, and instantiates the resources in a release
+
+Helm addresses use case 2.1: it provides a way to both create templates and
+search for/consume them.  Charts can also incorporate secrets, however, secrets
+created as part of a Helm release are not updated when the secret in the chart
```

Comment:
Secret generation #12732 is something we need to support, independent of Template, charts, whatever.